src/main/java/com/example/MachineService/controllers/MachineController.java

```diff
@@ -0,0 +1,55 @@
+package com.example.MachineService.controllers;
+
+import com.example.MachineService.entities.Machine;
+import com.example.MachineService.services.MachineService;
+import com.example.MachineService.services.UserService;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.core.Response;
+
+@RequestMapping("/tenant")
+@RestController
+public class MachineController {
+
+    private MachineService machineService;
+    private UserService userService;
+
+    public MachineController(MachineService machineService, UserService userService) {
+        this.machineService = machineService;
+        this.userService = userService;
+    }
+
+    @PostMapping("/{userId}/machine/device/")
+    Response addMachine(@RequestBody Machine machine, @PathVariable Long userId) {
```

Comment:
you should also catch the main exception and return a message instead(in all APIs)